cobra/solvers/coin.py

```diff
@@ -27,7 +27,7 @@ def objectiveValue_(self):
 
 
 def create_problem(cobra_model, objective_sense="maximize", **kwargs):
-    m = cobra_model.to_array_based_model()
+    m = cobra_model.copy().to_array_based_model()
```

Comment:
So it looks like I have to change this, otherwise calling the coin solver causes cobra_model.reaction._model to point to this new array_based_model. A `repair` call could work, but I don't really want to put in a conditional call to `model.repair()` if and only if the coin solver was used. Maybe there's a better way to handle this in `to_array_based_model`?